pkg/utils/gardener/shoot.go

```diff
@@ -647,19 +647,26 @@ func ConstructExternalDomain(ctx context.Context, c client.Reader, shoot *garden
 func ComputeRequiredExtensionsForShoot(shoot *gardencorev1beta1.Shoot, seed *gardencorev1beta1.Seed, controllerRegistrationList *gardencorev1beta1.ControllerRegistrationList, internalDomain, externalDomain *Domain) sets.Set[string] {
 	requiredExtensions := sets.New[string]()
 
-	if seed.Spec.Backup != nil {
+	if seed != nil && seed.Spec.Backup != nil {
 		requiredExtensions.Insert(ExtensionsID(extensionsv1alpha1.BackupBucketResource, seed.Spec.Backup.Provider))
 		requiredExtensions.Insert(ExtensionsID(extensionsv1alpha1.BackupEntryResource, seed.Spec.Backup.Provider))
 	}
```

Comment:
Are we now taking a short-cut by passing `seed == nil` or is this actually what we want long-term? I would have assumed that backup will (eventually) be necessary for autonomous shoot clusters as well. The same goes for the infrastructure/worker resources below. The seed provider type should actually be easy in an autonomous cluster as it should be the same for the shoot.
In summary, I would like to understand if you think this is just short-term workaround or the long-term solution?

Reply:
Very good question :) Yes, backup will be necessary. And yes, infra/worker will be necessary in the medium-touch case. However, I didn't want to overthink this already now and just focus on the problem we're solving at the moment: high-touch case. We can augment this later when we have time for it/come to it.